qiskit/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -141,50 +141,39 @@ def __init__(
         """
         validate_min("max_evals_grouped", max_evals_grouped, 1)
 
-        if sort_parameters_by_name is not None:
-            warnings.warn(
-                "The ``sort_parameters_by_name`` attribute is deprecated and will be "
-                "removed no sooner than 3 months after the release date of Qiskit Terra "
-                "0.18.0.",
-                DeprecationWarning,
-                stacklevel=2,
-            )
-
-        if ansatz is None:
-            ansatz = RealAmplitudes()
-
-        if optimizer is None:
-            optimizer = SLSQP()
-
-        if quantum_instance is not None:
-            if not isinstance(quantum_instance, QuantumInstance):
-                quantum_instance = QuantumInstance(quantum_instance)
-
         super().__init__()
 
+        # We don't use the setter for max_evals_grouped here, since it uses
+        # the getter for optimizer, and the setter for optimizer uses the
+        # getter for max_evals_grouped.
         self._max_evals_grouped = max_evals_grouped
         self._circuit_sampler = None  # type: Optional[CircuitSampler]
-        self._expectation = expectation
-        self._include_custom = include_custom
+        self._expectation = None
+        self.expectation = expectation
+        self._include_custom = None
+        self.include_custom = include_custom
```

Comment:
I don't think this is necessary for `include_custom` since it's not allowed to be None and therefore can also not be reset by setting it to None